R/data_manipulation.R

```diff
@@ -40,9 +40,9 @@ json_table_to_df <- function(dataset, n) {
 count_empty_desc <- function(table_df, table_colname) {
   table_df["empty"] <- NA
 
-  for (data_v in 1:nrow(table_df)) {
-    if ((nchar(table_df$description[data_v]) == 1) |
-      (table_df$description[data_v] == "Description to follow") |
+  for (data_v in seq_len(nrow(table_df))) {
```

Comment:
Suggestion that you do not have to tackle now since reviewers might have an opinion too.

Could the for loop be replaced with more functional code where the name of functions might help understand what's being calculated.

```
is_empty_description <- function(index, table) {
  nchar(table$description[index] ||
  table$description[index] == "Description to follow") ||
  table$description[index] == "NA" # why doesn't  `is.na()` work by the way?
}
# untested, and might be wrong
table$description <- purrr::map_lgl(seq_len(nrow(table_df)), is_empty_description, table = table)
# this returns a logical not character
```

Reply:
Related, recently I tried running _all_ linters in packages https://masalmon.eu/2024/08/28/lintr-3-steps/

Reply:
Ah, nice. Can I make this an Issue in the repo, if it does not need to be tackled now?

Reply:
yes of course

Reply:
https://github.com/aim-rsf/browseMetadata/issues/168